.circleci/config.yaml

```diff
@@ -0,0 +1,102 @@
+version: 2.1
+
+orbs:
+  python: circleci/python@2.1.1
+
+jobs:
+  setup:
+    executor:
+      name: python/default
+      tag: "3.10.15"
+    steps:
+      - checkout
+      - run:
+          name: Install uv
+          command: curl -LsSf https://astral.sh/uv/install.sh | sh
+      - persist_to_workspace:
+          root: /home/circleci
+          paths:
+            - "*"
+  test:
+    parameters:
+      package_subpath:
+        type: string
+    executor:
+      name: python/default
+      tag: "3.10.15"
+    steps:
+      - attach_workspace:
+          at: /home/circleci
+      - run:
+          name: Install deps
+          working_directory: /home/circleci/<< parameters.package_subpath >>
```

Comment:
This can be set once for the entire job above the `steps` section, e.g.,

```
version: 2.1
jobs:
  build:
    docker:
      - image: cimg/node:17.2.0
    environment:
      IMAGETAG: latest
    working_directory: ~/main
    steps:
      - run: echo "This is pipeline ID << pipeline.id >>"
```

Reply:
Also, the working_directory should be `/home/circleci/project/<< parameters.package_subpath >>

Reply:
i am not sure it can be set globally when it is a subpath of an attached workspace -- that is why i did it on each step rather than the global one.

noted on the path i will fix that